[enterprise-4.16] TELCODOCS-1861: Add Lifecycle Agent Enhancements 4.17 (#86758)
openshift/openshift-docs: modules/ztp-image-based-upgrade-procedure-steps.adoc

// Module included in the following assemblies:
// * edge_computing/image-based-upgrade/ztp-image-based-upgrade.adoc

:_mod-docs-content-type: CONCEPT
[id="ztp-image-based-upgrade-procedure-steps_{context}"]
= Performing an image-based upgrade on managed clusters at scale in several steps

> For use cases when you need better control of when the upgrade interrupts your service, you can upgrade a set of your managed clusters by using the `ImageBasedGroupUpgrade` CR with adding actions after the previous step is complete.

Reviewer: "previous step" might be tricky. What is the previous step. Best to call it out to avoid confusion.

Author: It's always a different step, so not sure how to better phrase it

> After evaluating the results of the previous steps, you can move to the next upgrade stage or troubleshoot any failed steps throughout the procedure.

Reviewer: Join the paragraph with the previous paragraph as this is how it shows on the preview.

Reviewer: What is the next upgrade stage. Best to state it explicitly.

Author: Same as above, it can be different depending on where you are in the flow

      - name: example-extra-manifests
        namespace: openshift-lifecycle-agent
    oadpContent: # <4>
      - name: oadp-cm
        namespace: openshift-adp
  plan: # <5>
    - actions: ["Prep"]
      rolloutStrategy:
        maxConcurrency: 2
        timeout: 2400
----
<1> Clusters to upgrade.
<2> Target platform version, the seed image to be used, and the secret required to access the image.
<3> Optional: Applies additional manifests, which are not in the seed image, to the target cluster. Also applies `ConfigMap` objects for custom catalog sources.
<4> `ConfigMap` resources that contain the OADP `Backup` and `Restore` CRs.
<5> Upgrade plan details.
--

. Apply created file by running the following command on the hub cluster:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Monitor the status updates by running the following command on the hub cluster:
+
--
[source,terminal]
----
$ oc get ibgu -o yaml
----

.Example output
[source,yaml]
----
# ...
status:
  clusters:
  - completedActions:
    - action: Prep
    name: spoke1
  - completedActions:
    - action: Prep
    name: spoke4
  - failedActions:
    - action: Prep
    name: spoke6
# ...
----

> The previous output of an example plan starts with the `Prep` stage only and you add actions to the plan based on the results of the previous step.

Reviewer: ```suggestion
The previous example starts with the `Prep` stage only and then adds actions to the plan based on the results of the previous step.
```
